src/protocol.rs

```diff
@@ -81,3 +81,17 @@ impl DerefMut for Protocol {
         &mut self.message
     }
 }
+
+#[inline(always)]
+pub fn timestamp_micros() -> u64 {
+    std::time::SystemTime::now()
```

Comment:
Since we are doing this, I would vote for avoiding SystemTime and using a monotonic source if possible.

Reply:
Agreed, the monotonic would be `std::Instant::now()`